src/main/java/com/marklogic/rest/util/RestTemplateUtil.java

```diff
@@ -40,22 +48,52 @@ public class RestTemplateUtil {
 		DEFAULT_CONFIGURERS.add(new UseSystemPropertiesConfigurer());
 	}
 
+	/**
+	 * @deprecated since 4.5.0; OkHttp is now the preferred client
+	 */
+	@Deprecated
 	public static RestTemplate newRestTemplate(String host, int port, String username, String password) {
 		return newRestTemplate(new RestConfig(host, port, username, password));
 	}
 
-	public static RestTemplate newRestTemplate(String host, int port, String username, String password, HttpClientBuilderConfigurer... configurers) {
-		return newRestTemplate(new RestConfig(host, port, username, password), configurers);
+	/**
+	 * As of 4.5.0, use this method for constructing a {@code RestTemplate} that supports all authentication types
+	 * supported by MarkLogic. ml-app-deployer is expected to use this for all calls as well starting in 4.5.0.
+	 *
+	 * @param config
+	 * @return
+	 */
+	public static RestTemplate newRestTemplate(RestConfig config) {
```

Comment:
@BillFarber  This is the critical change here - RestTemplate is now using an OkHttpClient built by the Java Client.